apps/homework/tasks.py

```diff
@@ -79,60 +102,68 @@ def post_submission(submission_pk, data_file=None):
-        os.environ.get('CODALAB_SUBMISSION_USERNAME'),
-        os.environ.get('CODALAB_SUBMISSION_PASSWORD')
-    ))
-    phases_dict = phases_request.json()[0]['phases']
-    phase_id = None
-    for phase in phases_dict:
-        if phase.get('is_active'):
-            phase_id = phase['id']
-            break
-
-    if not phase_id:
-        logger.info(f'No is_active field on phase. Update Codalab to latest version. Not POSTing a submission to Codalab.')
-        return
-
-    sub_descr = "Chagrade_Submission_{0}".format(submission.id)
-    finalize_url = "{0}/api/competition/{1}/submission?description={2}&phase_id={3}".format(site_url, challenge_pk,
-                                                                                            sub_descr, phase_id)
-    custom_filename = "{username}_{date}.zip".format(username=submission.creator.user.username, date=submission.created)
-    phase_final_resp = requests.post(finalize_url, data={
-        'id': submission_data,
-        'name': custom_filename,
-        'type': 'application/zip',
-    }, auth=HTTPBasicAuth(
-        os.environ.get('CODALAB_SUBMISSION_USERNAME'),
-        os.environ.get('CODALAB_SUBMISSION_PASSWORD')
-    ))
-
-    # If we succeed in posting to the phase, create a new tracker and store the submission info
-    if phase_final_resp.status_code == 201:
-        result = phase_final_resp.json()
-        new_tracker = SubmissionTracker.objects.create(
-            submission=submission,
-            remote_phase=phase_id,
-            remote_id=result['id'],
-            stored_status='Submitted'
-        )
-        logger.info(f'Succeeded posting submission with id {submission.id} to phase.')
-    else:
-        logger.info(f'Did not succeed in posting submission with id {submission.id} to phase.')
-    submission.submitted_to_challenge = True
+    if not submission.definition.jupyter_notebook_enabled:
+        # Example of url format we're expecting: https://competitions.codalab.org/api/competition/20616/phases/
+        phases_request_url = "{0}/api/competition/{1}/phases/".format(site_url, challenge_pk)
+        phases_request = requests.get(url=phases_request_url, auth=HTTPBasicAuth(
+            os.environ.get('CODALAB_SUBMISSION_USERNAME'),
+            os.environ.get('CODALAB_SUBMISSION_PASSWORD')
+        ))
+        phases_dict = phases_request.json()[0]['phases']
+        phase_id = None
+        for phase in phases_dict:
+            if phase.get('is_active'):
+                phase_id = phase['id']
+                break
+
+        if not phase_id:
+            logger.info(f'No is_active field on phase. Update Codalab to latest version. Not POSTing a submission to Codalab.')
+            return
+
+        sub_descr = "Chagrade_Submission_{0}".format(submission.id)
+        finalize_url = "{0}/api/competition/{1}/submission?description={2}&phase_id={3}".format(site_url, challenge_pk,
```

Comment:
newlines?

Reply:
Reformatted all the strings in this file to f-strings. ✅